src/user/ServiceManager.ts

```diff
@@ -301,7 +302,7 @@ class ServiceManager {
                     customDomain.indexOf(dotRootDomain) >= 0 &&
                     customDomain.indexOf(dotRootDomain) +
                         dotRootDomain.length ===
-                        customDomain.length
+                        appDomain.length
```

Comment:
This is not gonna work because it allows the user to choose a custom domain that could be used for another app.

For example:
- Your root domain: `root.domain.com`
- Your app: `app1`
- It allows the user to choose `app10.root.domain.com` as a custom domain.

Reply:
Fixed 2888903 & small refactor backed by test case